Updates to Contribute section

jasonblais opened the pull request:
This might be preliminary, but I started looking around the contribution section, updating a few of the pages and felt we're pretty close to being ready to ship this.

I've listed below the tasks that would be needed to ship it. Let me know if others are required.

Moreover, treat all of my changes as suggestions. I just brainstormed and prepared the changes, so I’m between 0/5 to 3/5 on most of these. Please be as critical as you'd like :P 

Also closes https://github.com/mattermost/mattermost-developer-documentation/issues/66

Punchlist to ship Contribute section live:

- [X] Separate contributor flow for each project 
  - [X] Condense contributor guidelines in Getting Started section, and turn it to a checklist
     - [X] Add a separate page for contributing without a Help Wanted ticket
  - [X] Create contributor guidelines for mattermost-server
  - [X] Create contributor guidelines for mattermost-webapp
  - [X] Create contributor guidelines for mattermost-redux
  - [X] Create contributor guidelines for mattermost-mobile
  - [X] Create contributor guidelines for desktop
- [X] Update Getting Started section
  - [X] Review intro
  - [X] Update core committers page
  - [X] Update project repositories
- [ ] Finish Desktop App section
  - [x] Update Developer environment setup with recent build changes
  - [ ] Add developer workflow 
- [x] Finish Mobile section
  - [x] Update intro
  - [x] Update Developer environment setup with recent build changes
  - [x] Add developer workflow 
- [X] Finish Server section
  - [x] Update intro
  - [X] Add developer workflow
- [X] Finish Webapp section
  - [x] Update intro
  - [X] Add developer workflow
- [X] Turn all pages public
- [X] Redirect dev docs from docs.mattermost.com to developers.mattermost.com https://github.com/mattermost/docs/pull/2037
  - [X] Contribution guidelines
  - [X] Developer flow
  - [ ] Developer environment setup
  - [ ] Mobile environment setup
  - [X] Core committers
- [ ] Rename platform binary to mattermost in all developer set up docs

jasonblais commented on site/content/contribute/getting-started/_index.md:
```diff
@@ -9,20 +8,24 @@ section: "contribute"
+3. **Set up your developer machine**. Each project has its own developer setup instructions.
 
-Unsure where you want to contribute? [Try starting with the server](/contribute/server/developer-setup/).
+4. **Select a [ticket](https://github.com/mattermost/mattermost-server/issues?q=is%3Aissue+is%3Aopen+label%3A%22Up+For+Grabs%22)** and comment to let people know you’re working on it. If there’s no ticket for what you want to work on, follow the process [here](https://docs.mattermost.com/developer/contribution-guide.html#pull-requests-for-minor-corrections-and-improvements-without-a-ticket-are-we ...
```
Fix link to /getting-started/contributions-without-ticket

jwilander commented on site/content/contribute/getting-started/_index.md:
```diff
@@ -1,5 +1,4 @@
 ---
-draft: true
 title: "Getting Started"
```
@jasonblais thoughts about making this page a lot simpler, similar to the integrate section?
https://developers.mattermost.com/integrate/getting-started/

jasonblais replied:
Great, I forgot we were already using that template. It'll be easy to migrate over to that.

Only question is: For step 2, I'm aiming to bring people right away to the project-specific contributor flow, with environment set up, etc. Should we keep that here, or how would we  go about it? 0/5

jwilander replied:
Yeah I think keeping it here makes sense

jwilander commented on site/content/contribute/getting-started/contribution-checklist.md:
```diff
@@ -0,0 +1,30 @@
+
+Thanks for your interest in contributing to Mattermost! Come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team.
+
+To help with translations, [see the localization process](https://docs.mattermost.com/developer/localization.html), follow this checklist for submitting a pull request (PR):
```
The first part of the sentence is about translations and the second part is about submitting a PR, seems a bit split-brained. Maybe divide them into two sentences and put them on different lines?

jasonblais replied:
Yeah, I'll fix that. It's confusing as-is.

jwilander commented on site/content/contribute/getting-started/contributions-without-ticket.md:
```diff
@@ -0,0 +1,23 @@
+ - [Fix Android app crashing when saving user notification settings](https://github.com/mattermost/mattermost-mobile/pull/364)
+ - [Fix recent mentions search not working](https://github.com/mattermost/mattermost-server/pull/5878)
+ 
+Larger pull requests, including those that require a change to documentation or user expectations, [require a Help Wanted issue opened by the core team](http://docs.mattermost.com/process/help-wanted.html) so that the change can be tested, documented and supported.
```
>tested, documented and supported

That sounds like the BS reason. We can do all that after a PR is submitted. The real reason to me is we want the product to have a unified vision, and to do that large changes need to discussed and agreed upon by the Mattermost core team.

The text that was there before is much more genuine IMO:

>Core committers who review the PR are entitled to reject it if they feel it significantly changes behavior or user expectations, which requires a Help Wanted ticket opened by the core team. This helps ensure that everything going into the project aligns with a unified vision.

jasonblais replied:
I just copy-pasted what we had. We can update that. Wasn't aware (or had forgotten) about the updated text.

jwilander replied:
Ah ok, I thought it was new text you had written 👍

jwilander commented on site/content/contribute/getting-started/respositories.md:
```diff
@@ -1,5 +1,4 @@
 ---
-draft: true
 title: "Project Respositories"
```
I'm unsure if we should keep this page or not. Don't know if it's helpful

jwilander commented:
Awesome work! This gets us real close to being to go live with this section

I added some comments and push some minor changes directly

jasonblais commented:
> This gets us real close to being to go live with this section

Is the rest captured in the punchlist in PR description, or is there something else I should add?

jwilander commented:
Nope, the PR description covers it

jasonblais commented:
@jwilander Wondering if you were interested taking over from here? Addressing your feedback in the PR (I agree with all of it) + queue a couple of tickets to finish the sections.

We can probably rally other people outside of ABC involved with the desktop/mobile sections. I'm not actually sure if there's much left for them, look pretty complete to me.

Thoughts?

I can finish the docs.mattermost.com stuff once the dev flows are completed.

jwilander commented:
@jasonblais I think is pretty much ready for review. I've made a bunch of updates and migrated all relevant pages from docs.mattermost.com. Only thing we're really missing is the desktop app workflow

jasonblais commented:
Awesome, let's discuss during our 1-1

yuya-oc commented on site/content/contribute/desktop/_index.md:
```diff
@@ -5,35 +5,16 @@ section: "contribute"
 
 # Desktop App
-The Mattermost desktop app is an [Electron](https://electronjs.org/) wrapper around the <a href="/contribute/webapp">webapp</a> project. It lives in the [mattermost/desktop](https://github.com/mattermost/desktop) repository. The desktop app runs on Windows, Linux, and macOS.
+The Mattermost desktop app is an [Electron](https://electronjs.org/) wrapper around the <a href="/contribute/webapp">webapp</a> project. It lives in the [mattermost/desktop]() repository. The desktop app runs on Windows, Linux, and macOS.
```
URL is missing.

`[mattermost/desktop]()` -> `[mattermost/desktop](https://github.com/mattermost/desktop)`

jwilander replied:
Nice catch, will update

yuya-oc commented:
https://github.com/mattermost/mattermost-developer-documentation/blob/a38bdcf03a95167a014468ab597958c0df009755/site/content/contribute/desktop/developer-setup.md#prerequisites

On "Prerequisites" section, Node.js should be upgraded: `Node.js 8 or later`.

jwilander commented:
Thanks @yuya-oc! Updated
